modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/SimpleTypeDataQualifierVisitor.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.introspection;
+
+import org.mule.extension.introspection.DataQualifier;
+import org.mule.extension.introspection.DataQualifierVisitor;
+
+/**
+ * Base implementation for a {@link DataQualifierVisitor} which adds the new
+ * method {@link #onSimpleType()} which is invoked by all the {@link DataQualifier}s
+ * which refer to a simple type. Those would be:
+ * <p/>
+ * <ul>
+ * <li>{@link #onBoolean()}</li>
+ * <li>{@link #onInteger()}</li>
+ * <li>{@link #onDouble()}</li>
+ * <li>{@link #onDecimal()}</li>
+ * <li>{@link #onString()}</li>
+ * <li>{@link #onLong()}</li>
+ * <li>{@link #onEnum()}</li>
+ * </ul>
+ * <p/>
+ * All other qualifiers delegate into {@link #defaultOperation()} by default, but they
+ * can be overridden at will
+ *
+ * @since 3.7.0
+ */
+public abstract class SimpleTypeDataQualifierVisitor extends BaseDataQualifierVisitor
+{
+
+    protected abstract void onSimpleType();
+
+    @Override
+    public void onBoolean()
+    {
+        onSimpleType();
```

Comment:
This is a convenience class for cases in which the code to execute when finding a simpleType and the code on defaultOperation() is different. Look at the implementations of this class and it should make sense